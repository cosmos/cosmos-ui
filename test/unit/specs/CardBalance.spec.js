import Vue from 'vue'
import CardBalance from '../../../app/src/renderer/components/wallet/CardBalance'

describe('CardBalance', () => {
  let Cmp, vm
  let propsData = {
    balance: {
      denom: 'jbcoin',
      amount: 1234.56
    }
  }

  beforeEach(() => {
    Cmp = Vue.extend(CardBalance)
    vm = new Cmp({ propsData: propsData }).$mount()
  })

  it('has the expected html structure', () => {
    expect(vm.$el).toMatchSnapshot()
  })
})
