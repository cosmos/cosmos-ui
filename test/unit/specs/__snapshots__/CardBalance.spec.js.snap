// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardBalance has the expected html structure 1`] = `
<div
  class="card-balance"
>
  <div
    class="card-balance-container"
  >
    <div
      class="left"
    >
      <div
        class="title"
      >
        <i
          class="fa fa-circle-o"
        />
        <span>
          JBCOIN
        </span>
      </div>
      <div
        class="balance"
      >
        <div
          class="unit"
        >
          Balance:
        </div>
        <div
          class="value"
        >
          <span
            class="integer"
          >
            1234
          </span>
          <span
            class="fraction"
          >
            .55999
          </span>
        </div>
      </div>
      <div
        class="actions"
      >
        <button
          class="ni-btn"
        >
          <span
            class="ni-btn-container"
          >
            <i
              aria-hidden="true"
              class="ni-btn-icon material-icons"
            >
              search
            </i>
            <span
              class="ni-btn-value"
            >
              Details
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
